### PR TITLE
Enable tf.nn.bias_add python op tests to work in eager mode (as well as graph mode)

### PR DESCRIPTION
**Pull request by duncanriach**
This current PR is a follow-on from [PR 31465](https://github.com/tensorflow/tensorflow/pull/31465) (Add GPU-deterministic tf.nn.bias_add). This current PR enhances the testing of the tf.nn.bias_add op to include eager as well as graph mode.

### CODE REVIEW

**Review by chsigg — DISMISSED** [review]

**Comment by gbaned** [comment]
@duncanriach  Can you please resolve conflicts? Thanks!

**Comment by duncanriach** [comment]
@gbaned, that was a pretty hairy rebase involving manual resolution of conflicts with changes made by @sanjoy yesterday, in which he added a `parameterized` decorator from `absl.testing`.

I'm pretty sure I haven't broken anything (but I still have to run tests locally). The only issue I have now is that @sanjoy's changes prevented a work-around I was using in `tensorflow/python/kernel_tests/bias_op_deterministic_test.py` where it's not possible (currently) to use a `cached_session` with `force_gpu=True` for testing in both eager and graph mode. In the original pull request, before the `parameterized` decorator was added, I worked-around this by setting a non-cached session context, with `force_gpu=True`, outside the nested loops that provided the parameter combinations. With the `parameterized` decorator, I can no longer set the session context at that level.

@sanjoy, please will you have a look at this for me and see if it's easy for you to suggest a better way of doing it?

**Comment by sanjoy** [comment]
> where it's not possible (currently) to use a `cached_session` with `force_gpu=True` for testing in both eager and graph mode.

Why do you need to use the cached session?  If that's only to make the test run faster then I'd say we can leave the change as is.

**Comment by duncanriach** [comment]
@sanjoy: That's right. Just to make the test run faster. If you're happy with how this is, then so am I.
@chsigg: Please will you re-review this?

**Comment by sanjoy** [comment]
Looks like this is waiting on @chsigg 

(Gaurav, sorry for the notification spam.)

**Comment by duncanriach** [comment]
History of this PR:
* Nov 21, 2019: approved by @chsigg
* Nov 21, 2019: CI revealed a tiny pylint error, which I resolved the same day. Re-review then pended.
* Dec 10, 2019: I was informed of a conflict with another merge, which I resolved on the same day. Re-review then pended.
* Dec 26, 2019: @sanjoy removed @chsigg re-review request, and seems to have left it without and requested review.

@gbaned, please will you help me to get this reviewed and merged? Should it, or can it, be assigned to another, or an additional, reviewer?

**Comment by sanjoy** [comment]
I'll ping @chsigg internally to see if he can do another review or I'll find another reviewer.

**Review by chsigg — DISMISSED** [review]

**Comment by gbaned** [comment]
@duncanriach Could you please address Ubuntu Sanity errors? Thanks!

**Comment by duncanriach** [comment]
Fixed Ubuntu Sanity error, then ran sanity locally. @chsigg, please re-approve following [this change](https://github.com/tensorflow/tensorflow/pull/33803/commits/4ea10c4bcc1ca3d98e34c6742220c2c8fe9df946).

**Review by sanjoy — APPROVED** [review]
Blanket approval since there are only whitespace changes after the previous approval.

**Comment by duncanriach** [comment]
Hi @sanjoy, this PR has been `ready to pull` since January 10 (two months ago). Is there any way it can be moved closer to merge?

**Comment by rthadur** [comment]
@mihaimaruseac can you please help pull this PR manually.

**Comment by mihaimaruseac** [comment]
Done

File: tensorflow/python/kernel_tests/bias_op_base.py [hunk 1]
```diff
@@ -20,17 +20,22 @@
...
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import gradient_checker
+from tensorflow.python.ops import gradient_checker_v2
 from tensorflow.python.ops import gradients_impl
 from tensorflow.python.ops import nn_ops
 import tensorflow.python.ops.nn_grad  # pylint: disable=unused-import
 from tensorflow.python.platform import test
 
 
+@test_util.run_all_in_graph_and_eager_modes
```
Comment (rthadur) [inline tensorflow/python/kernel_tests/bias_op_base.py]: `Missing class docstring [missing-docstring]` can you please check this error ?

**Pull request merged**